src/config_file.c

```diff
@@ -1056,36 +1056,38 @@ static int parse_conditional_include(git_config_parser *reader,
 static int read_on_variable(
 	git_config_parser *reader,
 	const char *current_section,
-	char *var_name,
-	char *var_value,
+	const char *var_name,
+	const char *var_value,
 	const char *line,
 	size_t line_len,
 	void *data)
 {
 	struct parse_data *parse_data = (struct parse_data *)data;
 	git_buf buf = GIT_BUF_INIT;
 	cvar_t *var;
+	char *name;
 	int result = 0;
 
 	GIT_UNUSED(line);
 	GIT_UNUSED(line_len);
 
-	git__strtolower(var_name);
-	git_buf_printf(&buf, "%s.%s", current_section, var_name);
-	git__free(var_name);
+	name = git__strdup(var_name);
+	GITERR_CHECK_ALLOC(name);
 
-	if (git_buf_oom(&buf)) {
-		git__free(var_value);
+	git__strtolower(name);
+	git_buf_printf(&buf, "%s.%s", current_section, name);
+	git__free(name);
```

Comment:
It seems disappointing to have to do an extra tiny `malloc` here for `name` n every variable that we read and immediately `free` it.  Doing a little bit more manipulation and not doing a `printf` here might help.  eg:

```
git_buf_puts(&buf, current_section);
git_buf_putc(&buf, '.');

for (c = name, *c; c++)
    git_buf_putc(&buf, git__tolower(*c));
```

Reply:
Agreed. Your solution looks a lot cleaner, thanks!